packages/core/src/codewhisperer/service/transformByQ/transformProjectValidationHandler.ts

```diff
@@ -70,72 +66,9 @@ async function getMavenJavaProjects(javaProjects: TransformationCandidateProject
-            const spawnResult = spawnSync(baseCommand, args, { shell: false, encoding: 'utf-8' })
-            if (spawnResult.status !== 0) {
-                let errorLog = ''
-                errorLog += spawnResult.error ? JSON.stringify(spawnResult.error) : ''
-                errorLog += `${spawnResult.stderr}\n${spawnResult.stdout}`
-                getLogger().error(`CodeTransformation: Error in running javap command = ${errorLog}`)
-                let errorReason = ''
-                if (spawnResult.stdout) {
-                    errorReason = 'JavapExecutionError'
-                } else {
-                    errorReason = 'JavapSpawnError'
-                }
-                if (spawnResult.error) {
-                    const errorCode = (spawnResult.error as any).code ?? 'UNKNOWN'
-                    errorReason += `-${errorCode}`
-                }
-                getLogger().error(
-                    `CodeTransformation: Error in running javap command = ${errorReason}, log = ${errorLog}`
-                )
-            } else {
-                const majorVersionIndex = spawnResult.stdout.indexOf('major version: ')
-                const javaVersion = spawnResult.stdout.slice(majorVersionIndex + 15, majorVersionIndex + 17).trim()
-                if (javaVersion === CodeWhispererConstants.JDK8VersionNumber) {
-                    detectedJavaVersion = JDKVersion.JDK8
-                } else if (javaVersion === CodeWhispererConstants.JDK11VersionNumber) {
-                    detectedJavaVersion = JDKVersion.JDK11
-                } else {
-                    detectedJavaVersion = JDKVersion.UNSUPPORTED
-                }
-            }
-        }
-
-        // detectedJavaVersion will be undefined if there are no .class files or if javap errors out, otherwise it will be JDK8, JDK11, or UNSUPPORTED
-        project.JDKVersion = detectedJavaVersion
-        projectsValidToTransform.push(project)
-    }
-    return projectsValidToTransform
-}
-
-/*
- * This function filters all open projects by first searching for a .java file and then searching for a pom.xml file in all projects.
- * It also tries to detect the Java version of each project by running "javap" on a .class file of each project.
- * As long as the project contains a .java file and a pom.xml file, the project is still considered valid for transformation,
- * and we allow the user to specify the Java version.
- */
+// This function filters all open projects by first searching for a .java file and then searching for a pom.xml file in all projects.
 export async function validateOpenProjects(projects: TransformationCandidateProject[]) {
     const javaProjects = await getJavaProjects(projects)
-
     const mavenJavaProjects = await getMavenJavaProjects(javaProjects)
-
-    // These projects we know must contain a pom.xml and a .java file
-    const projectsValidToTransform = await getProjectsValidToTransform(mavenJavaProjects)
```

Comment:
Deleted this function entirely because 1) it was expensive (running a shell command (`javap`) on every open project) and 2) it was just used to format a String (see `createLanguageUpgradeConfirmationPrompt` above); every project returned by `getMavenJavaProjects` above was also returned by `getProjectsValidToTransform`